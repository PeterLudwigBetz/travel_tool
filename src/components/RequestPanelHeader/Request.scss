.request-panel-header {
  box-sizing: border-box;
  background-color: #F8F8F8;
  height: 201.9px;
  width: 96.3%;
  @media (max-width: 479px){
    height: 150.9px;
    width: 94%;
}
  .open-requests {
    display: flex;
    justify-content: space-between;
    align-items: center;
    position: relative;
    top: 50px;
    height: 36px;
  }
}
