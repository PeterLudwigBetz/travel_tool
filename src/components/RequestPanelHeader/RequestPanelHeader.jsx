import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import PageHeader from '../PageHeader';
import ButtonGroup from '../button-group/ButtonGroup';
import HeaderPagination from '../Pagination/HeaderPagination';
import './Request.scss';

class RequestPanelHeader extends PureComponent {
  renderButtonGroup = () => {
    const {
      openRequestsCount,
      fetchUserRequests,
      pastRequestsCount,
      limit
    } = this.props;
    return (
      <ButtonGroup
        openRequestsCount={openRequestsCount}
        fetchUserRequests={fetchUserRequests}
        pastRequestsCount={pastRequestsCount}
        limit={limit}
        buttonsType="requests"
      />
    );
  };

  render() {
    const { requests, getRequestsWithLimit, toggleNewRequestModal } = this.props;
    return (
      <div className="request-panel-header">
        <PageHeader title="REQUESTS" actionBtn="New Request" toggleNewRequestModal={toggleNewRequestModal} />
        {requests &&
          requests.length > 0 && (
          <div className="open-requests">
            {this.renderButtonGroup()}
            <HeaderPagination
              getRequestsWithLimit={getRequestsWithLimit}
            />
          </div>
        )}
      </div>
    );
  }
}

RequestPanelHeader.propTypes = {
  openRequestsCount: PropTypes.number,
  pastRequestsCount: PropTypes.number,
  limit: PropTypes.number.isRequired,
  getRequestsWithLimit: PropTypes.func.isRequired,
  fetchUserRequests: PropTypes.func.isRequired,
  requests: PropTypes.array,
  toggleNewRequestModal: PropTypes.func.isRequired,
};

RequestPanelHeader.defaultProps = {
  openRequestsCount: null,
  pastRequestsCount: null,
  requests: null,
};

export default RequestPanelHeader;
