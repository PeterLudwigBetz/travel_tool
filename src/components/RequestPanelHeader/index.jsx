import React, { PureComponent } from 'react';
import Header from '../Header';
import ButtonGroup from '../ButtonGroup';
import HeaderPagination from '../Pagination/HeaderPagination';
import './Request.scss';

class RequestPanelHeader extends PureComponent {
  render() {
    return (
      <div className="request-panel-header">
        <Header />
        <div className="open-requests">
          <ButtonGroup />
          <HeaderPagination />
        </div>
      </div>
    );
  }
}

export default RequestPanelHeader;
