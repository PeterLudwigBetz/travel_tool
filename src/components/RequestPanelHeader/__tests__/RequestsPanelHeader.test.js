import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import RequestPanelHeader from '../RequestPanelHeader';

const props = {
  openRequestsCount: 1,
  pastRequestsCount: 0,
  limit: 10,
  getRequestsWithLimit: sinon.spy(() => Promise.resolve()),
  fetchUserRequests: sinon.spy(() => Promise.resolve()),
  requests: [
    {
      id: 'xDh20btGz',
      name: 'Amarachukwu Agbo',
      origin: 'Lagos',
      destination: 'Nairobi',
      manager: 'Samuel Kubai',
      gender: 'Female',
      department: 'TDD',
      role: 'Software Developer',
      status: 'Open',
      userId: 'pommyLHJmKrx76A8Slm',
      departureDate: '2018-12-09',
      arrivalDate: '2018-12-11',
    },
  ]
};

describe('<RequestPanelHeader />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<RequestPanelHeader {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
