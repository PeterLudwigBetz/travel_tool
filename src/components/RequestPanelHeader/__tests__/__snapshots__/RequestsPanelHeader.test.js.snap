// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestPanelHeader /> should render correctly 1`] = `
<div
  className="request-panel-header"
>
  <PageHeader
    actionBtn="New Request"
    title="REQUESTS"
    toggleNewRequestModal={[Function]}
  />
  <div
    className="open-requests"
  >
    <ButtonGroup
      buttonsType="requests"
      fetchUserRequests={[Function]}
      limit={10}
      openRequestsCount={1}
      pastRequestsCount={0}
    />
    <HeaderPagination
      getRequestsWithLimit={[Function]}
    />
  </div>
</div>
`;
